File: easy/go/2215-find-the-difference-of-two-arrays.go
package main

func findDifference(nums1 []int, nums2 []int) [][]int {
 nums1Map := getMap(nums1)
 nums2Map := getMap(nums2)

 onlyInNums1 := []int{}
 onlyInNums2 := []int{}

 for k, _ := range nums1Map {
	_, ok := nums2Map[k]
	if !ok {
		onlyInNums1 = append(onlyInNums1, k)
	}
 }

 for k, _ := range nums2Map {
	_, ok := nums1Map[k]
	if !ok {
		onlyInNums2 = append(onlyInNums2, k)
	}
 }

 return [][]int{onlyInNums1, onlyInNums2}
}

func getMap(nums []int) map[int]bool {
	numMap := make(map[int]bool);
	for i := 0; i < len(nums); i++ {
		_, ok := numMap[nums[i]]
		if !ok {
			numMap[nums[i]] = true	
		}
	}

	return numMap
}

File: easy/javascript/2215-find-the-difference-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function (nums1, nums2) {
  let onlyInNums1 = [];
  let onlyInNums2 = [];

  const num1Set = new Set(nums1);
  const num2Set = new Set(nums2);

  for (const v of num1Set) {
    if (!num2Set.has(v)) {
      onlyInNums1.push(v);
    }
  }

  for (const v of num2Set) {
    if (!num1Set.has(v)) {
      onlyInNums2.push(v);
    }
  }

  return [onlyInNums1, onlyInNums2];
};
